test/luatest_helpers/vtest.lua

```diff
@@ -523,9 +523,17 @@ local function router_cfg(router, cfg)
 end
 
 --
--- Create a new router in the cluster.
+-- Apply the config on the given router inside fiber
 --
-local function router_new(g, name, cfg)
+local function router_fiber_cfg(router, cfg)
```

Comment:
Lets not complicate the API too much and reuse the existing `router_new` function. In particular, lets make so that `router_new` won't call `router_cfg` if `cfg` is nil. Then you can make empty `router_new` and then call `router_cfg` in your test in a fiber (the fiber doesn't need to be on the router instance, it can be local to the test instance too - might be simpler).

Reply:
That's a great idea! Implemented.